### ISSUE TITLE
CA1823 raised correctly for partial classes, but disappears once opening the file

### ISSUE THREAD

**Issue by adraut** [dotnet/roslyn-analyzers#1546]
#### Analyzer package
Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer
AvoidUnusedPrivateFieldsAnalyzer

#### Repro steps
1. Create a C# full framework project.
2. Add the analyzers packages.
3. Add a file like the below to the project with 2 partial class declarations
```
    public partial class C1
    {
        private readonly string str = "";
    }

    public partial class C1
    {
    }
```
4. Close the file.
5. Build.
6. Observe that the diagnostic/warning `Unused field 'str'` is in the build output and the Error List window in Visual Studio.
7. Open the file.
8. Observe that the warning disappears from the Error List and that the diagnostic is no longer raised.

The analyzer is not triggered for non-partial class.  I did not try any other scenarios.

#### Expected behavior
The diagnostic is raised during compilation and while the file is open.

#### Actual behavior
The diagnostic is raised during compilation, but not while the file is open.

**Comment by mavasani** [comment]
This is due to VS IDE not running compilation end actions when FSA is off.

See https://github.com/dotnet/roslyn-analyzers/issues/1559#issuecomment-363196049, whose root cause is https://github.com/dotnet/roslyn/issues/11750.

Tagging @heejaechang @jinujoseph @kuhlenh, we are going to keep getting these reports until https://github.com/dotnet/roslyn/issues/11750 is addressed.

**Comment by mavasani** [comment]
Closing this as external.
